test/newwint.F90

```diff
@@ -0,0 +1,62 @@
+! This is a test for NCEPLIBS-bufr.
+!
+! This is a test for the newwin() subroutine.
+!
+! Ed Hartnett, J. Woollen, 3/15/2023
+program newwint
+  use moda_usrint
+  use moda_tables
+!  implicit none
+  integer idate, ins1, ins2, iret, iwin, jwin, lun, m, n
+  character*8 subset
+  integer msg_num, NUM_MSG
+  parameter(NUM_MSG = 12)
+  real(8) ncdt(5,50)
+  common /usrstr/ nnod,ncon,nods(20),nodc(10),ivls(10),kons(10)
+
+  print *,'Testing newwin()...'
+  
+  ! open a file 
+  open(unit = 20, file = 'testfiles/IN_2', form = 'UNFORMATTED')
+  call openbf(20,'IN',20)
+
+  ! Read each message in the file.
+  do msg_num = 1, NUM_MSG
+     iret = ireadmg(20,subset,idate)
```

Comment:
When ireadmg() is called for the first time, it returns 0.

So this test is not working.

How many messages can we expect in this test file? The test should not be written to handle any number of messages (as the code origianlly did), it should know exactly how many messages to expect from the test data file, and fail if it gets anything other than the exact correct number of messages.

Similarly for everything else in the file. We want the test to be brittle, not robust. If something in the code or test files changes, we want the test to break, not adapt.

Reply:
@jack-woollen and @jbathegit comments and answers welcome.